src/storage/src/storage_state.rs

```diff
@@ -1196,13 +1196,13 @@ impl StorageState {
         match cmd {
             StorageCommand::CreateTimely { .. } => panic!("CreateTimely must be captured before"),
             StorageCommand::InitializationComplete => (),
-            StorageCommand::UpdateConfiguration(mut params) => {
+            StorageCommand::UpdateConfiguration(params) => {
                 // These can be done from all workers safely.
                 tracing::info!("Applying configuration update: {params:?}");
 
                 // We serialize the dyncfg updates in StorageParameters, but configure
                 // persist separately.
-                if let Some(updates) = params.dyncfg_updates.take() {
```

Comment:
@guswynn this bug prevented the `ALTER SYSTEM SET storage_enforce_external_addresses = true` from working. I think we were accidentally taking the option value from the updates which meant it didn't get propagated  to the `InternalStorageCommand::UpdateConfiguration` command

Reply:
Can you break this out in its own pr please? This is a affects a critical parameter (`storage_dataflow_delay_sources_past_rehydration`) and we need to merge this fix before the release cut today

Sorry about this!

Reply:
yes can do!

Reply:
https://github.com/MaterializeInc/materialize/pull/26189